sceptre/stack.py

```diff
@@ -520,28 +520,57 @@ def create_change_set(self, change_set_name):
 
         :param change_set_name: The name of the change set.
         :type change_set_name: str
+        :raises: botocore.exceptions.ClientError
         """
-        create_change_set_kwargs = {
-            "StackName": self.external_name,
-            "Parameters": self._format_parameters(self.parameters),
-            "Capabilities": ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
-            "ChangeSetName": change_set_name,
-            "NotificationARNs": self.config.get("notifications", []),
-            "Tags": [
-                {"Key": str(k), "Value": str(v)}
-                for k, v in self.config.get("stack_tags", {}).items()
-            ]
-        }
-        create_change_set_kwargs.update(self._get_template_details())
-        create_change_set_kwargs.update(self._get_role_arn())
-        self.logger.debug(
-            "%s - Creating change set '%s'", self.name, change_set_name
-        )
-        self.connection_manager.call(
-            service="cloudformation",
-            command="create_change_set",
-            kwargs=create_change_set_kwargs
-        )
+        for change_set_type in ["UPDATE", "CREATE"]:
+            create_change_set_kwargs = {
+                "StackName": self.external_name,
+                "Parameters": self._format_parameters(self.parameters),
+                "Capabilities": ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
+                "ChangeSetName": change_set_name,
+                "NotificationARNs": self.config.get("notifications", []),
+                "ChangeSetType": change_set_type,
+                "Tags": [
+                    {"Key": str(k), "Value": str(v)}
+                    for k, v in self.config.get("stack_tags", {}).items()
+                ]
+            }
+            create_change_set_kwargs.update(self._get_template_details())
+            create_change_set_kwargs.update(self._get_role_arn())
+            self.logger.debug(
+                "%s - Creating %s change set '%s'",
+                self.name, change_set_type, change_set_name
+            )
+            try:
+                self.connection_manager.call(
+                    service="cloudformation",
+                    command="create_change_set",
+                    kwargs=create_change_set_kwargs
+                )
+            except botocore.exceptions.ClientError as e:
+                # If a stack does not exist, describe_stacks will throw an
+                # exception. Unfortunately we don't have a better way than
+                # parsing the exception msg to understand the nature of this
+                # exception.
+                msg = str(e)
+
+                if (
+                        "Stack with id {0} does not exist".format(
```

Comment:
Could be a bit looser with the condition e.g. `msg.endswith("does not exist")` - you can also get the message from the exception via `msg = exp.response["Error"]["Message"]`

Reply:
Since this is forked code from awscli, I minimized the differences beyond style. I typically favor this approach since this is noted in the commit message for these lines, and if we need to update this in the future when these things change, it's useful to be able to draw from updated awscli source.